core/new-gui/src/app/workspace/component/property-editor/property-editor.component.ts

```diff
@@ -112,12 +136,48 @@ export class PropertyEditorComponent {
     // hide the view first and then make everything null
     this.displayForm = false;
 
+    this.showAdvanced = false;
     this.currentOperatorID = undefined;
     this.currentOperatorInitialData = undefined;
     this.currentOperatorSchema = undefined;
-
+    this.tempOperatorSchema = undefined;
   }
 
+  /**
+   * Hides the advanced option fields in the operator by deleting the field.
+   * @param operator
+   */
+  public hideAdvancedFormLayout(operatorSchema: OperatorSchema | undefined): OperatorSchema {
+    if (! operatorSchema) {
+      throw new Error('Parameter operator schema is undefined');
+    }
+    this.tempOperatorSchema = cloneDeep(this.currentOperatorSchema);
+    if (! this.currentOperatorSchema) {
+      throw new Error('Current operator schema is undefined');
+    }
+    const advancedOptionsList = this.currentOperatorSchema.additionalMetadata.advancedOptions;
+    if (! advancedOptionsList) {
+      return this.currentOperatorSchema;
+    }
+    // make a deep clone of the operator schema and change its properties
+    const jsonSchemaToModify = cloneDeep(operatorSchema.jsonSchema);
+    advancedOptionsList.forEach(advancedOption => {
+      if (jsonSchemaToModify.properties && advancedOption in jsonSchemaToModify.properties) {
+        delete jsonSchemaToModify.properties[advancedOption];
+      }
+      // handles the scenario where the advanced option is also required
+      if (jsonSchemaToModify.required && jsonSchemaToModify.required.includes(advancedOption)) {
+          const index: number = jsonSchemaToModify.required.indexOf(advancedOption);
+          if (index !== -1) {
+          jsonSchemaToModify.required.splice(index, 1); }
+        }
+    });
```

Comment:
Here, when you create the new, modified json schema, you uses the `delete` object property operation defined in JavaScript. However, according to this stackoverflow post, https://stackoverflow.com/questions/208105/how-do-i-remove-a-property-from-a-javascript-object, the `delete` operation is extremely slow. 

Instead, using spread operator seems to be a better choice here, which we will not directly modify the original object.

Here is the proposed implementation using spread `...` operator:
```
    const currentSchema = operatorSchema.jsonSchema;
    let currentSchemaProperties = currentSchema.properties;
    const currentSchemaRequired = cloneDeep(currentSchema.required);

    advancedOptionsList.forEach(
      advancedOption => {
        if (currentSchemaProperties && advancedOption in currentSchemaProperties) {
          const { [advancedOption] : [], ...newProperties} = currentSchemaProperties;
          currentSchemaProperties = newProperties;
        }

        if (currentSchemaRequired && currentSchemaRequired.includes(advancedOption)) {
          const index = currentSchemaRequired.indexOf(advancedOption);
          if (index !== -1) {
            currentSchemaRequired.splice(index, 1);
          }
        }
      }
    );
```